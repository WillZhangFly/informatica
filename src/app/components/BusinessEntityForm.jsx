import React, { useState, useEffect } from "react";
import { useSelector } from "react-redux";
import TextInput from "./TextInput";

function BusinessEntityForm() {
  let originValue = useSelector((state) => state.entity);

  const [entity, setEntity] = useState(useSelector((state) => state.entity));
  const [saveResult, setSaveResult] = useState({});
  const normalizeMeta = useSelector((state) => {
    return state.meta.field.filter((_metaField) => entity[_metaField.name]);
  });

  console.log("compare", originValue, entity);

  const saveEntity = (e) => {
    e.preventDefault();
    setSaveResult({
      ...entity,
      $original: {
        ...originValue,
      },
    });
  };
  const handleInputChange = (e) => {
    setEntity({
      ...entity,
      [e.target.id]: e.target.value,
    });
  };

  return (
    <div className="justify-content-center d-flex">
      <div className="card p-3 col-6 mt-5">
        <h2>Business Entity</h2>
        <form onSubmit={saveEntity}>
          <div className="form-group row">
            {normalizeMeta.map((_field) => {
              return (
                <TextInput
                  handleChange={handleInputChange}
                  value={entity[_field.name]}
                  key={_field.name}
                  name={_field.label}
                  id={_field.name}
                  dataType={_field.dataType}
                />
              );
            })}
          </div>
          <button
            type="submit"
            className="form-control mt-2 btn btn-primary col-4"
          >
            Save
          </button>
          <div className="mt-4">
            <h2>Saved results print here:</h2>
            {JSON.stringify(saveResult)};
          </div>
        </form>
      </div>
    </div>
  );
}

export default BusinessEntityForm;
