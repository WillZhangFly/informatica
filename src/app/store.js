import { configureStore } from "@reduxjs/toolkit";
import entityReducer from "../features/entity/entitySlice";
import logger from "redux-logger";

export default configureStore({
  reducer: {
    entity: entityReducer,
  },
  middleware: [logger],
});
