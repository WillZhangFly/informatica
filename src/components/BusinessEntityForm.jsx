import React from "react";

function BusinessEntityForm() {
  return (
    <div className="justify-content-center d-flex">
      <div className="card p-3 col-6 mt-5">
        <h2>Business Entity</h2>
        <form onSubmit={() => {}}>
          <div class="form-group row">
            <label for="firstName" class="col-sm-2 col-form-label">
              First Name
            </label>
            <div class="col-sm-4">
              <input type="text" class="form-control" id="firstName" />
            </div>
            <label for="lastName" class="col-sm-2 col-form-label">
              Last Name
            </label>
            <div class="col-sm-4">
              <input type="text" class="form-control" id="lastName" />
            </div>
          </div>
          <button type="submit" className="form-control mt-2 btn btn-primary">
            Save
          </button>
        </form>
      </div>
    </div>
  );
}

export default BusinessEntityForm;
