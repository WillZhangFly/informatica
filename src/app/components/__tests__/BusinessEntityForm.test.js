import React from "react";
import { render, screen, fireEvent } from "../../utils/test-utils";
import BusinessEntityForm from "../BusinessEntityForm";
import { Provider } from "react-redux";
import store from "../../store";

import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe("<BusinessEntityForm />", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = render(
      <Provider store={store}>
        <BusinessEntityForm />
      </Provider>
    );
  });

  it("should render", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("renders without crashing", () => {
    fireEvent.click(screen.getByRole("button"), {});
    expect(screen.queryByText(/WINTON/)).toBeInTheDocument();
  });
});
