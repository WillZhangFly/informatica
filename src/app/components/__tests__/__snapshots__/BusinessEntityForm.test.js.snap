// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BusinessEntityForm /> should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="justify-content-center d-flex"
      >
        <div
          class="card p-3 col-6 mt-5"
        >
          <h2>
            Business Entity
          </h2>
          <form>
            <div
              class="form-group row"
            >
              <label
                class="col-sm-2 col-form-label mt-3"
                for="partyType"
              >
                Party Type
              </label>
              <div
                class="col-sm-4"
              >
                <input
                  class="form-control mt-3"
                  id="partyType"
                  type="text"
                  value="Person"
                />
              </div>
              <label
                class="col-sm-2 col-form-label mt-3"
                for="lastName"
              >
                Last Name
              </label>
              <div
                class="col-sm-4"
              >
                <input
                  class="form-control mt-3"
                  id="lastName"
                  type="text"
                  value="HARLEY"
                />
              </div>
              <label
                class="col-sm-2 col-form-label mt-3"
                for="displayName"
              >
                Display Name
              </label>
              <div
                class="col-sm-4"
              >
                <input
                  class="form-control mt-3"
                  id="displayName"
                  type="text"
                  value="WILLIAM WINTON HARLEY"
                />
              </div>
              <label
                class="col-sm-2 col-form-label mt-3"
                for="generationSuffixCd"
              >
                Generation Suffix Cd
              </label>
              <div
                class="col-sm-4"
              >
                <input
                  class="form-control mt-3"
                  id="generationSuffixCd"
                  type="text"
                  value="III"
                />
              </div>
              <label
                class="col-sm-2 col-form-label mt-3"
                for="middleName"
              >
                Middle Name
              </label>
              <div
                class="col-sm-4"
              >
                <input
                  class="form-control mt-3"
                  id="middleName"
                  type="text"
                  value="WINTON"
                />
              </div>
              <label
                class="col-sm-2 col-form-label mt-3"
                for="birthdate"
              >
                Birthdate
              </label>
              <div
                class="col-sm-4"
              >
                <input
                  class="form-control mt-3"
                  id="birthdate"
                  type="date"
                  value="1886-09-07"
                />
              </div>
              <label
                class="col-sm-2 col-form-label mt-3"
                for="taxId"
              >
                Tax ID
              </label>
              <div
                class="col-sm-4"
              >
                <input
                  class="form-control mt-3"
                  id="taxId"
                  type="text"
                  value="1234567  "
                />
              </div>
              <label
                class="col-sm-2 col-form-label mt-3"
                for="firstName"
              >
                First Name
              </label>
              <div
                class="col-sm-4"
              >
                <input
                  class="form-control mt-3"
                  id="firstName"
                  type="text"
                  value="WILLIAM"
                />
              </div>
              <label
                class="col-sm-2 col-form-label mt-3"
                for="rowidObject"
              >
                Rowid Object
              </label>
              <div
                class="col-sm-4"
              >
                <input
                  class="form-control mt-3"
                  id="rowidObject"
                  type="text"
                  value="1242          "
                />
              </div>
            </div>
            <button
              class="form-control mt-2 btn btn-primary col-4"
              type="submit"
            >
              Save
            </button>
            <div
              class="mt-4"
            >
              <h2>
                Saved results print here:
              </h2>
              {}
              ;
            </div>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="justify-content-center d-flex"
    >
      <div
        class="card p-3 col-6 mt-5"
      >
        <h2>
          Business Entity
        </h2>
        <form>
          <div
            class="form-group row"
          >
            <label
              class="col-sm-2 col-form-label mt-3"
              for="partyType"
            >
              Party Type
            </label>
            <div
              class="col-sm-4"
            >
              <input
                class="form-control mt-3"
                id="partyType"
                type="text"
                value="Person"
              />
            </div>
            <label
              class="col-sm-2 col-form-label mt-3"
              for="lastName"
            >
              Last Name
            </label>
            <div
              class="col-sm-4"
            >
              <input
                class="form-control mt-3"
                id="lastName"
                type="text"
                value="HARLEY"
              />
            </div>
            <label
              class="col-sm-2 col-form-label mt-3"
              for="displayName"
            >
              Display Name
            </label>
            <div
              class="col-sm-4"
            >
              <input
                class="form-control mt-3"
                id="displayName"
                type="text"
                value="WILLIAM WINTON HARLEY"
              />
            </div>
            <label
              class="col-sm-2 col-form-label mt-3"
              for="generationSuffixCd"
            >
              Generation Suffix Cd
            </label>
            <div
              class="col-sm-4"
            >
              <input
                class="form-control mt-3"
                id="generationSuffixCd"
                type="text"
                value="III"
              />
            </div>
            <label
              class="col-sm-2 col-form-label mt-3"
              for="middleName"
            >
              Middle Name
            </label>
            <div
              class="col-sm-4"
            >
              <input
                class="form-control mt-3"
                id="middleName"
                type="text"
                value="WINTON"
              />
            </div>
            <label
              class="col-sm-2 col-form-label mt-3"
              for="birthdate"
            >
              Birthdate
            </label>
            <div
              class="col-sm-4"
            >
              <input
                class="form-control mt-3"
                id="birthdate"
                type="date"
                value="1886-09-07"
              />
            </div>
            <label
              class="col-sm-2 col-form-label mt-3"
              for="taxId"
            >
              Tax ID
            </label>
            <div
              class="col-sm-4"
            >
              <input
                class="form-control mt-3"
                id="taxId"
                type="text"
                value="1234567  "
              />
            </div>
            <label
              class="col-sm-2 col-form-label mt-3"
              for="firstName"
            >
              First Name
            </label>
            <div
              class="col-sm-4"
            >
              <input
                class="form-control mt-3"
                id="firstName"
                type="text"
                value="WILLIAM"
              />
            </div>
            <label
              class="col-sm-2 col-form-label mt-3"
              for="rowidObject"
            >
              Rowid Object
            </label>
            <div
              class="col-sm-4"
            >
              <input
                class="form-control mt-3"
                id="rowidObject"
                type="text"
                value="1242          "
              />
            </div>
          </div>
          <button
            class="form-control mt-2 btn btn-primary col-4"
            type="submit"
          >
            Save
          </button>
          <div
            class="mt-4"
          >
            <h2>
              Saved results print here:
            </h2>
            {}
            ;
          </div>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
